### PR TITLE
Support RMN in tooling

### PR DESCRIPTION
**Pull request by connorwstein**
Add a new NewLocalDevEnvironmentWithRMN which creates RMN containers in addition to the rest of the environment. TestRMN is a placeholder and can be run locally.

### CODE REVIEW

File: integration-tests/deployment/devenv/rmn_test.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestExtractPeerID(t *testing.T) {
+	data := []byte("{\"Salt\":\"yAUVSNGPnY78hJwJwKDBmA==\",\"Nonce\":\"8p0vNKpnQH1P+7/cg2dM3vNlc60tzPl0\",\"Ciphertext\":\"VboRchQzDx/+zIVDWyTXIEwo4Ej0s7O6kQqmgCqW+A+HEXl6bI02W1Y2T88XuY0m44eC9DvSpBPhs/VLtgymj6nBcl+nzfJHLMp2pBMPjKHxgNsEk34mDgnmqYKTtIkgzv7hEyy7j0CAcR/RxkjfQNKpfWOlNtAFryFO+w==\",\"AdditionalData\":\"{\\\"PeerID\\\":\\\"12D3KooWNqugYSJw9thwwu1PC3aEpPsBxMZM2EdzpJXGesRG4E8n\\\"}\"}")
+	peerID, err := extractPeerID(data)
+	require.NoError(t, err)
```
Comment (AnieeG) [inline integration-tests/deployment/devenv/rmn_test.go:13]:
```suggestion
	require.NoError(t, err)
	require.Equal(t, peerID, "12D3KooWNqugYSJw9thwwu1PC3aEpPsBxMZM2EdzpJXGesRG4E8n")
```

File: integration-tests/deployment/devenv/rmn.go [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+	}
+
+	l := tc.Logger
+	if t != nil {
+		l = logging.CustomT{
+			T: t,
+			L: lggr,
+		}
+	}
+	keystore := "keystore/rageproxy-keystore.json"
```
Comment (AnieeG) [inline integration-tests/deployment/devenv/rmn.go]: how about moving this to constant?

**Review by AnieeG — COMMENTED** [review]
need to fix lint

File: integration-tests/deployment/ccip/test_helpers.go [hunk 3]
```diff
@@ -292,7 +297,133 @@ func NewLocalDevEnvironment(t *testing.T, lggr logger.Logger) DeployedEnv {
...
+		"latest",
+		"afn2proxy",
+		"latest",
+		dockerenv.LogStream,
+	)
+	require.NoError(t, err)
+	return tenv, *rmnCluster
+}
+
+func NetworksToRPCMap(evmNetworks []*blockchain.EVMNetwork) map[uint64]string {
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/test_helpers.go]:
```suggestion
func MustNetworksToRPCMap(evmNetworks []*blockchain.EVMNetwork) map[uint64]string {
```

File: integration-tests/deployment/ccip/rmn_test.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package ccipdeployment
+
+import (
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func TestRMN(t *testing.T) {
```
Comment (AnieeG) [inline integration-tests/deployment/ccip/rmn_test.go:9]: can we move this to smoke/ccip_test.go ? It will be easier to trigger this in CI

Comment (connorwstein) [reply integration-tests/deployment/ccip/rmn_test.go:9]: probably can once its complete. Won't be ready for CI for a while

**Review by AnieeG — DISMISSED** [review]

**Review by AnieeG — APPROVED** [review]

**Review by ogtownsend — APPROVED** [review]

File: integration-tests/deployment/ccip/rmn_test.go (hunk 4 above)
Comment (kalverra) [reply integration-tests/deployment/ccip/rmn_test.go:9]: So is it unfinished? Or just unable to run in CI? Mind updating the `t.Skip()` with a more descriptive message?

**Review by kalverra — CHANGES_REQUESTED** [review]

**Review by kalverra — APPROVED** [review]

**Pull request merged**